x/mongo/driver/topology/server.go

```diff
@@ -107,9 +107,14 @@ type Server struct {
 	subscriptionsClosed bool
 }
 
+// updateTopologyCallback is a callback used to create a server that should be called when the parent Topology instance
+// should be updated based on a new server description. The callback must return the server description that should be
+// stored by the server.
+type updateTopologyCallback func(description.Server) description.Server
```

Comment:
This was a convenience change. There are many places where we were declaring such a function so having a dedicated type made sense.